scripts/dts/gen_defines.py

```diff
@@ -101,6 +107,7 @@ def main():
             write_existence_flags(node)
 
             out_dev(node, "ZDID", node_to_zdid[node])
+            print("extern struct device __device_{};".format(node_to_zdid[node]), file=dev_extern_file)
```

Comment:
It would be nice if this include some identification:
```
...
extern struct device __device_4;   /* SILABS_GETCKO_I2C#0 */
```
or something else so we could catch the declaration name using grep.  I'm imagining cases where I have a pointer, look it up in a linker map, and want to know what it belongs to.